plugins/module_utils/mongodb_common.py

```diff
@@ -25,6 +25,23 @@
     pymongo_found = False
 
 
+def autoinstall_pymongo(module):
+    installed = False
+    if module.check_mode:
+        module.fail_json(msg="pymongo must be installed to use check mode. "
+                             "If run normally this module can auto-install it.")
+    try:
+        module.warn("Attempting installation of pymongo via pip.")
+        module.run_command(['/usr/bin/pip3', 'install', 'pymongo'], check_rc=True)
```

Comment:
This needs to use `ansible_python_interpreter` somehow. Ansible can be run from a virtualenv. Plus there may be more than one python3 install, and so it is important to use the same one that ansible is using. There's also a question of using `--user` in some situations.

Reply:
Using `/usr/bin/pip3` also requires `become=true`, so we might need a check for permissions before attempting to install with a particular pip binary.

Reply:
Hello,

This was inspired by the apt module that installs its own python driver when required. Sadly these modules are architected different and that made this difficult. There were also a bunch of issues with different containers, with different python versions, hence the hardcoded paths to get a particular container working. It basically turned into a bit of a hackathon and then other priorities came up. This is wayxdown the list and I'lm mindful to kick it off since I think there's more important stuff to be getting on with.

Cheers,

Rhys